src/worker/actions.ts

```diff
@@ -1,265 +1,32 @@
-      const vitest = vitestById[id]
-      const files = await globTestFiles(id, vitest)
-      // reset cached test files list
-      vitest.projects.forEach((project) => {
-        project.testFilesList = null
-      })
-      return files.map(([project, spec]) => [project.config.name || '', spec])
-    },
-    async isTestFile(file: string) {
-      for (const [_, vitest] of vitestEntries) {
-        for (const project of vitest.projects) {
-          if (project.isTestFile(file))
-            return true
-        }
-      }
-      return false
-    },
-    async enableCoverage(id: string) {
-      const vitest = vitestById[id]
-      coverages.delete(vitest)
-      vitest.config.coverage.enabled = true
-      coverageStatuses.set(vitest, true)
-
-      const jsonReporter = vitest.config.coverage.reporter.find(([name]) => name === 'json')
-      vitest.config.coverage.reporter = jsonReporter ? [jsonReporter] : [['json', {}]]
-      vitest.config.coverage.reportOnFailure = true
-      vitest.config.coverage.reportsDirectory = join(tmpdir(), `vitest-coverage-${randomUUID()}`)
-
-      try {
-        if (!providers.has(vitest)) {
-          // @ts-expect-error private method
-          await vitest.initCoverageProvider()
-          await vitest.coverageProvider?.clean(vitest.config.coverage.clean)
-          providers.set(vitest, vitest.coverageProvider!)
-        }
-        else {
-          vitest.coverageProvider = providers.get(vitest)
-          await vitest.coverageProvider?.clean(vitest.config.coverage.clean)
-        }
-      }
-      catch (err) {
-        coverageStatuses.set(vitest, false)
-        vitest.config.coverage.enabled = false
-        throw err
-      }
-    },
-    disableCoverage(id: string) {
-      const vitest = vitestById[id]
-      coverages.delete(vitest)
-      coverageStatuses.set(vitest, false)
-      vitest.config.coverage.enabled = false
-      vitest.coverageProvider = undefined
-    },
-    async getCoverageConfig(id: string) {
-      const vitest = vitestById[id]
-      return vitest.config.coverage
-    },
-    async waitForCoverageReport(id: string) {
-      const vitest = vitestById[id]
-      const coverage = vitest.config.coverage
-      if (!coverage.enabled || !vitest.coverageProvider)
-        return null
-      await vitest.runningPromise
-      if (existsSync(coverage.reportsDirectory))
-        return coverage.reportsDirectory
-      return null
-    },
-    startInspect(port) {
-      inspector().open(port)
-      debuggerEnabled = true
-    },
-    stopInspect() {
-      debuggerEnabled = false
-      inspector().close()
-    },
+export function createWorkerMethods(vitestById: Record<string, Vitest>): VitestPool {
+  return new Proxy<any>({
     async close() {
-      for (const vitest of vitestEntries) {
+      for (const vitest in vitestById) {
         try {
-          await vitest[1].close()
+          await vitestById[vitest].dispose()
         }
         catch {
           // ignore
```

Comment:
When can this happen? Is it worth adding logging here? Or guard against it instead, and "seal it off" with tests?

Reply:
https://github.com/vitest-dev/vitest/issues/2008